### PR TITLE
feat: github workflow to publish the `backend-plugin-manager` to NPMJS in the `@janus-idp` scope

### PR DESCRIPTION
**Pull request by davidfestal**
GitHub workflow to publish the `backend-plugin-manager` to NPMJS in the `@janus-idp` scope

This fixes issue https://github.com/janus-idp/backstage-showcase/issues/525

### CODE REVIEW

File: .github/workflows/publish-backend-plugin-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+          LAST_PUSHED_PRERELEASE_WITHOUT_NUMBER=$(echo $LAST_PUSHED_VERSION | cut -d- -f2 | cut -d. -f1)
+          LAST_PUSHED_PRERELEASE_NUMBER=$(echo $LAST_PUSHED_VERSION | cut -d- -f2 | cut -d. -f2)
+
+          if [ "$COMMIT_MAJOR.$COMMIT_MINOR.$PATCH" == "$LAST_PUSHED_MAJOR.$LAST_PUSHED_MINOR.$LAST_PUSHED_PATCH" ]; then
+              PRERELEASE="janus.$((LAST_PUSHED_PRERELEASE_NUMBER+1))"
+          else
+              PRERELEASE="janus.0"
+          fi
+          echo "VERSION=$COMMIT_MAJOR.$COMMIT_MINOR.$PATCH-$PRERELEASE" >> "$GITHUB_OUTPUT"
+          echo "GH Output: " && cat $GITHUB_OUTPUT
```
Comment (tumido) [inline .github/workflows/publish-backend-plugin-manager.yaml:110]: Would you mind documenting this code a bit? It's a bit complicated and confusing. 🙂

**Review by tumido — COMMENTED** [review]
Left few suggestions. All are just nits, nothing blocking.

File: .github/workflows/publish-backend-plugin-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+# limitations under the License.
+
+name: Publish Backend Plugin Manager
+
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+env:
+  SCOPE: ${{ vars.BACKEND_PLUGIN_MANAGER_SCOPE }}
```
[the remark this answers is not in the document]
Comment (davidfestal) [reply .github/workflows/publish-backend-plugin-manager.yaml:22]: That helped testing it on my branch until it is merged.

File: .github/workflows/publish-backend-plugin-manager.yaml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+          sed -i "s/\"version\": \".*\"/\"version\": \"$VERSION\"/" packages/backend-plugin-manager/package.json
+          sed -i "s/\"name\": \"@backstage\/backend-plugin-manager\"/\"name\": \"$SCOPE\/backend-plugin-manager\"/" packages/backend-plugin-manager/package.json
+          sed -i "s/\"private\": true/\"private\": false/" packages/backend-plugin-manager/package.json
+          echo "Package.JSON:" && cat packages/backend-plugin-manager/package.json
+
+      - name: Install Dependencies
+        env:
+          YARN_ENABLE_IMMUTABLE_INSTALLS: 'false'
+        if: steps.test_commit_already_published.outputs.ALREADY_PUSHED == 'false'
+        run: yarn install
```
[the remark this answers is not in the document]
Comment (davidfestal) [reply .github/workflows/publish-backend-plugin-manager.yaml]: even `--prefer-offline` doesn't work? I get this error:

```
YN0050: The --prefer-offline flag is deprecated; use the --cached flag with 'yarn add' instead
```

I'd rather keep a simple `yarn install` TBF

File: .github/workflows/publish-backend-plugin-manager.yaml [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          repository: backstage/backstage
+          ref: master
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 18
```
Comment (tumido) [inline .github/workflows/publish-backend-plugin-manager.yaml:43]:
```suggestion
          node-version-file: '.nvmrc'
          cache: 'yarn'
```

File: .github/workflows/publish-backend-plugin-manager.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tumido) [reply .github/workflows/publish-backend-plugin-manager.yaml:154]: I think we should couple `prefer-offline` with cache (as suggested below)

File: .github/workflows/publish-backend-plugin-manager.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tumido) [reply .github/workflows/publish-backend-plugin-manager.yaml:24]: Sure, can we remove it now? 🙂

**Comment by davidfestal** [comment]
@tumido I fixed the review comments that has a fix, and answered to the other ones, which should not be blockers afaict.

File: .github/workflows/publish-backend-plugin-manager.yaml (hunk 4 above)
Comment (tumido) [reply .github/workflows/publish-backend-plugin-manager.yaml:52]: Good call on `.nvmrc`. 👍 Can we still enable the cache?

Comment (davidfestal) [reply .github/workflows/publish-backend-plugin-manager.yaml:52]: I added `cache: 'yarn'` as you requested.

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**